src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1503,7 +1485,17 @@ internal bool GetExternAliasTarget(string aliasName, out NamespaceSymbol @namesp
         /// </summary>
         internal new NamedTypeSymbol? ScriptClass
         {
-            get { return _scriptClass.Value; }
+            get
+            {
+                // _scriptClass is allowed to be null, thus why a bool is used to track initialization
+                if (!_scriptClassInitialized)
+                {
+                    _scriptClassInitialized = true;
+                    Interlocked.CompareExchange(ref _scriptClass, BindScriptClass(), null);
```

Comment:
❗ This pattern could allow a concurrent caller to observe an uninitialized value

Reply:
Reordering the bool assignment and the CompareExchange should address this, right?

Reply:
➡️ Continuing this discussion in https://github.com/dotnet/roslyn/pull/72371#discussion_r1511738117